ethcore/src/client/client.rs

```diff
@@ -293,11 +318,12 @@ impl<V> Client<V> where V: Verifier {
 
 		{
 			if !good_blocks.is_empty() && self.block_queue.queue_info().is_empty() {
+				let (enacted, retracted) = self.calculate_enacted_retracted(routes);
 				io.send(NetworkIoMessage::User(SyncMessage::NewChainBlocks {
 					good: good_blocks,
 					bad: bad_blocks,
-					// TODO [todr] were to take those from?
-					retracted: vec![],
+					enacted: enacted,
```

Comment:
`good` - blocks that have been successfuly inserted to blockchain (doesn't matter if canon / fork)
`bad` - blocks that were in queue but haven't been inserted to blockchain (because of validation)
`enacted` - blocks which after batch inserting all `good` are (at the end) in canon chain
`retracted` - blocks which after batch inserting all `good` are (at the end) in non-canon chain.